build-plugin/src/main/kotlin/thunderbird.quality.badging.gradle.kts

```
@@ -17,51 +17,63 @@ plugins {
+            val generateBadging = tasks.register<GenerateBadgingTask>(generateBadgingTaskName) {
+                apk.set(variant.artifacts.get(SingleArtifact.APK_FROM_BUNDLE))
+                aapt2Executable.set(
+                    File(
+                        android.sdkDirectory,
+                        "${SdkConstants.FD_BUILD_TOOLS}/" +
+                            "${android.buildToolsVersion}/" +
+                            SdkConstants.FN_AAPT2,
+                    ),
+                )
+                badging.set(
+                    project.layout.buildDirectory.file(
+                        "outputs/apk_from_bundle/${variant.name}/${variant.name}-badging.txt",
+                    ),
+                )
+            }
 
-        val updateBadgingTaskName = "update${capitalizedVariantName}Badging"
-        tasks.register<Copy>(updateBadgingTaskName) {
-            from(generateBadging.get().badging)
-            into(project.layout.projectDirectory.dir("badging"))
-        }
+            val updateBadgingTaskName = "update${capitalizedVariantName}Badging"
+            tasks.register<Copy>(updateBadgingTaskName) {
+                from(generateBadging.get().badging)
+                into(project.layout.projectDirectory.dir("badging"))
+            }
 
-        val checkBadgingTaskName = "check${capitalizedVariantName}Badging"
-        val goldenBadgingPath = project.layout.projectDirectory.file("badging/${variant.name}-badging.txt")
-        tasks.register<CheckBadgingTask>(checkBadgingTaskName) {
-            if (goldenBadgingPath.asFile.exists()) {
-                goldenBadging.set(goldenBadgingPath)
+            val checkBadgingTaskName = "check${capitalizedVariantName}Badging"
+            val goldenBadgingPath = project.layout.projectDirectory.file("badging/${variant.name}-badging.txt")
+            tasks.register<CheckBadgingTask>(checkBadgingTaskName) {
+                if (goldenBadgingPath.asFile.exists()) {
+                    goldenBadging.set(goldenBadgingPath)
+                }
+                generatedBadging.set(
+                    generateBadging.get().badging,
+                )
+                this.updateBadgingTaskName.set(updateBadgingTaskName)
+
+                output.set(
+                    project.layout.buildDirectory.dir("intermediates/$checkBadgingTaskName"),
+                )
             }
-            generatedBadging.set(
-                generateBadging.get().badging,
-            )
-            this.updateBadgingTaskName.set(updateBadgingTaskName)
 
-            output.set(
-                project.layout.buildDirectory.dir("intermediates/$checkBadgingTaskName"),
-            )
+            tasks.named("build") {
+                dependsOn(checkBadgingTaskName)
+            }
         }
     }
 }
 
+private fun String.capitalized() = replaceFirstChar {
+    if (it.isLowerCase()) it.titlecase(Locale.getDefault()) else it.toString()
+}
```

We definitely do not want our task names to be locale-specific. See https://en.wikipedia.org/wiki/Dotted_and_dotless_I_in_computing

The deprecated method contains a `ReplaceWith` annotation that does the right thing.

```suggestion
private fun String.capitalized() = replaceFirstChar {
    if (it.isLowerCase()) it.titlecase() else it.toString()
}
```

---

This is just copy paste from https://github.com/android/nowinandroid/blob/main/build-logic/convention/src/main/kotlin/com/google/samples/apps/nowinandroid/Badging.kt

But yes, it doesn't make sense.

---

I submitted a PR to the nowinandroid repo: https://github.com/android/nowinandroid/pull/1644